src/ws/cockpitcertificate.c

```diff
@@ -111,8 +134,24 @@ openssl_make_dummy_cert (const gchar *key_file,
   gint exit_status;
   gchar *stderr_str = NULL;
   gchar *command_line = NULL;
+  gchar *ssl_config = NULL;
   gchar *subject = generate_subject ();
 
+  /* make config file with subjectAltName for localhost and our tests */
+  ssl_config = create_temp_file (g_get_tmp_dir (), "ssl.conf.XXXXXX", error);
+  if (!ssl_config)
+      return FALSE;
+  if (!g_file_set_contents (ssl_config,
+              "[ req ]\n"
+              "req_extensions = v3_req\n"
+              "extensions = v3_req\n"
+              "distinguished_name = req_distinguished_name\n"
+              "[ req_distinguished_name ]\n"
+              "[ v3_req ]\n"
+              "subjectAltName=IP:127.0.0.1,IP:127.0.0.2,DNS:*.cockpit.lan\n",
```

Comment:
This seems very targeted to out tests. These self-signed certs are generated everywhere not just for tests.

i think localhost and 127.0.0.1 but I'm not sure we should be issuing certs for cockpit.lan. If we need cockpit.lan for tests we can always generate a certificate for that in our test setup code.

Reply:
Agree here. The change is good, but likely another change needed for the test.

@martinpitt Kudos for figuring this out.

Reply:
Ah yes, good point. How about `IP:127.0.0.1,DNS:localhost` for a starter? If there is demand, we can expand the list later on for reasonable and common items, such as for 192.168.*.

After finding the right kind of logging in chromium, I found out the [reason](https://bugs.chromium.org/p/chromium/issues/detail?id=757181) for my woes. I have a solution in mind now, and it's completely unrelated to this PR, so this is both low prio and also "if you have doubts, then nevermind". But at least for localhost it would be a nice improvement, as that's a common case.

Reply:
Pushed the "localhost only" change.